### ISSUE TITLE
Add more information about symfony specific use cases

### ISSUE THREAD

**Issue by icanhazstring** [composer-unused/composer-unused#394]
Right now its possible that composer-unused might report certain symfony packages as unused.
This comes down to the fact that symfony uses several other way to place `.php` files.

For example:
- `public/`
- `var/cache/`
- `config/`

To kind of mitigate unused packages we can add a part in the [README.md](README.md) to add some information.

```
return static function (Configuration $config): Configuration
{
    return $config
        ->setAdditionalFilesFor('shopping/s2-customer-delivery', [
            __FILE__,
            ...Glob::glob(__DIR__ . '/var/cache/dev/Container*/*.php'),
            ...Glob::glob(__DIR__ . '/config/**/*.php'),
            ...Glob::glob(__DIR__ . '/public/*.php'),
        ]);
};
```

**Comment by lookyman** [comment]
I feel like `setAdditionalFilesFor` is not enough for Symfony, because of the per-dependency configuration. If you want to add `config/budles.php`, you have to exclude every single foreign referenced bundle.

**Comment by icanhazstring** [comment]
No. config/bundles.php is part of your application. So you want composer-unused to scan this file as part of your application.

Your application also counts as a package, the root package.

**Comment by lookyman** [comment]
> This only has to be done, if your package contains a file which is not served through means of the composer.json of that package.

Which is exactly my use case. I have a bunch of packages that add stuff that is actually used, and another bunch that provide integration to Symfony via bundles. The only place those are referenced is in config/bundles.php. So I am stuck with adding configuration for each of them?

Btw sorry about spamming maybe unrelated issue here..

**Comment by icanhazstring** [comment]
> Which is exactly my use case. I have a bunch of packages that add stuff that is actually used, and another bunch that provide integration to Symfony via bundles. The only place those are referenced is in config/bundles.php. So I am stuck with adding configuration for each of them?

If the bundle is referenced using the FQN in config/bundles.php you just add this file as an additional file with your application name (composer.json->name) as explained in the initial message.

This way composer-unused will scan this file, see the used symbols and will mark the package as used.

**Comment by lookyman** [comment]
Oh NOW I get it, thanks!

**Comment by icanhazstring** [comment]
Yes. That is the 2nd level requirement which can't be detected right now. I hope I find time to implement that.

**Comment by Hanmac** [comment]
i noticed that this doesn't even help because `twig/extra-bundle` doesn't have the `twig/*-extra` libs as requirement (or suggestion) in composer.json

maybe that's even another problem 🤔

**Comment by icanhazstring** [comment]
Yes that is also possible. But the code has symbols used from that package. Having that would mark it required.

**Comment by ph-il** [comment]
The exemple for twig might been also flawed. If you don't use any extensions in your twig files, you don't used them even if you refer to them.  

The others one, is notifier, as extra package will be set in the yaml, and sometime only for production. Same as messager.

**Comment by ph-il** [comment]
I think you need plug-in for differents projects/framework. A little like phpstan.
